algorithms/kernel/pca/transform/oneapi/pca_transform_dense_default_batch_oneapi_impl.i

```diff
@@ -0,0 +1,316 @@
+    auto build_options = fptype_name;
+    ExecutionContextIface & ctx    = services::Environment::getInstance()->getDefaultExecutionContext();
+    ClKernelFactoryIface & factory = ctx.getClKernelFactory();
+
+    const services::String options = getKeyFPType<algorithmFPType>();
+    services::String cachekey("__daal_algorithms_pca_transform");
+    cachekey.add(fptype_name);
+    factory.build(ExecutionTargetIds::device, cachekey.c_str(), pca_transform_cl_kernels, build_options.c_str());
+
+    const char * const whiteningKernel = "whitening";
+    KernelPtr kernel = factory.getKernel(whiteningKernel);
+
+    const unsigned int workItemsPerGroup = (numComponents > maxWorkItemsPerGroup) ? maxWorkItemsPerGroup : numComponents;
+    KernelArguments args(4);
+    args.set(0, transformedBlock, AccessModeIds::readwrite);
+    args.set(1, invEigenvalues, AccessModeIds::read);
+    args.set(2, maxWorkItemsPerGroup);
+    args.set(3, numComponents);
+
+    KernelRange local_range(workItemsPerGroup);
+    KernelRange global_range(workItemsPerGroup * numVectors);
+
+    KernelNDRange range(1);
+    range.global(global_range, &status);
+    DAAL_CHECK_STATUS_VAR(status);
+    range.local(local_range, &status);
+    DAAL_CHECK_STATUS_VAR(status);
+
+    ctx.run(range, kernel, args, &status);
+
+    return status;
+}
+
+
+template<typename algorithmFPType, transform::Method method>
+services::Status TransformKernelOneAPI<algorithmFPType, method>::allocateBuffer(ExecutionContextIface& context,
+                                                                                UniversalBuffer& returnBuffer,
+                                                                                uint32_t bufferSize)
+{
+    services::Status status;
+
+    const algorithmFPType zero = 0.0;
+    returnBuffer = context.allocate(TypeIds::id<algorithmFPType>(), bufferSize, &status);
+    DAAL_CHECK_STATUS_VAR(status);
+    context.fill(returnBuffer, zero, &status);
+
+    return status;
+}
+
+template<typename algorithmFPType, transform::Method method>
+services::Status TransformKernelOneAPI<algorithmFPType, method>::copyBuffer(ExecutionContextIface& context,
+                                                                            UniversalBuffer& returnBuffer,
+                                                                            NumericTable* data,
+                                                                            uint32_t nRows, uint32_t nCols)
+
+{
+    services::Status status;
+
+    BlockDescriptor<algorithmFPType> dataBlock;
+    DAAL_CHECK_STATUS(status, data->getBlockOfRows(0, nRows, ReadWriteMode::readOnly, dataBlock));
+    context.copy(returnBuffer, 0, dataBlock.getBuffer(), 0, nRows * nCols, &status);
+    data->releaseBlockOfRows(dataBlock);
+
+    return status;
+}
+
+template<typename algorithmFPType, transform::Method method>
+services::Status TransformKernelOneAPI<algorithmFPType, method>::copyBufferByRef(ExecutionContextIface& context,
```

Comment:
What the difference between `copyBuffer` and `copyBufferByRef`? I think you have to keep only one method, where `data` is passed as a reference. You can use it for both cases